File: lena/utils/find.py
import os
import sys

def find_files(directory, extension):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(extension):
                print(os.path.join(root, file))


if __name__ == "__main__":
    if len(sys.argv) > 2:
        directory = sys.argv[1]
        extension = sys.argv[2]
        find_files(directory, extension)

File: lena/scripts/find.js
const fs = require('fs');
const path = require('path');

function findFiles(directory, extension) {
	fs.readdir(directory, { withFileTypes: true }, (err, files) => {
		if (err) {
			console.log(err);
			return;
		}
		files.forEach(file => {
			if (file.isDirectory()) {
				findFiles(path.join(directory, file.name), extension);
			} else if (file.name.endsWith(extension)) {
				console.log(path.join(directory, file.name));
			}
		});
	});
}

const args = process.argv.slice(2);

if (args.length > 1) {
	const directory = args[0];
	const extension = args[1];
	findFiles(directory, extension);
} else {
	console.log('[lena-i->scripts]: Usage: node find.js <directory> <extension>');
}
